src/pybind/ceph_daemon.py

```diff
@@ -0,0 +1,275 @@
+        `last_dump`.
+        """
+        val_row = ""
+        for section_name, names in self._stats.items():
+            for stat_name, stat_nick in names.items():
+                stat_type = self._schema[section_name][stat_name]['type']
+                if bool(stat_type & COUNTER):
+                    n = max(dump[section_name][stat_name] -
+                            last_dump[section_name][stat_name], 0)
+                elif bool(stat_type & LONG_RUNNING_AVG):
+                    entries = dump[section_name][stat_name]['avgcount'] - \
+                            last_dump[section_name][stat_name]['avgcount']
+                    if entries:
+                        n = (dump[section_name][stat_name]['sum'] -
+                             last_dump[section_name][stat_name]['sum']) \
+                            / float(entries)
+                        n *= 1000.0  # Present in milliseconds
+                    else:
+                        n = 0
+                else:
+                    n = dump[section_name][stat_name]
+
+                val_row += self.format_dimless(n, self.col_width(stat_nick))
+                val_row += " "
+            val_row = val_row[0:-1]
+            val_row += self.colorize("|", self.BLUE)
+        val_row = val_row[0:-len(self.colorize("|", self.BLUE))]
+        ostr.write("{0}\n".format(val_row))
+
+    def _load_schema(self):
+        """
+        Populate our instance-local copy of the daemon's performance counter
+        schema, and work out which stats we will display.
+        """
+        self._schema = json.loads(admin_socket(self.asok_path, ["perf", "schema"]))
+
+        # Build list of which stats we will display, based on which
+        # stats have a nickname
+        self._stats = defaultdict(dict)
+        for section_name, section_stats in self._schema.items():
+            for name, schema_data in section_stats.items():
+                if schema_data['nick']:
+                    self._stats[section_name][name] = schema_data['nick']
+
+    def run(self, ostr=sys.stdout):
+        """
+        Print output at regular intervals until interrupted.
+
+        :param ostr: Stream to which to send output
+        """
+
+        self._load_schema()
+        self._colored = self.supports_color(ostr)
+
+        self._print_headers(ostr)
+
+        last_dump = json.loads(admin_socket(self.asok_path, ["perf", "dump"]))
+        rows_since_header = 0
+        term_height = 25
+        while True:
+            dump = json.loads(admin_socket(self.asok_path, ["perf", "dump"]))
+            if rows_since_header > term_height - 2:
+                self._print_headers(ostr)
+                rows_since_header = 0
+            self._print_vals(ostr, dump, last_dump)
+            rows_since_header += 1
+            last_dump = dump
+            try:
+                time.sleep(1)
+            except KeyboardInterrupt:
```

Comment:
seems to me that this `KeyboardInterrupt` should catch the whole function. Otherwise, when sending a `CTRL-C` when it is *not* doing the `time.sleep` it will break with a traceback